Glyssen/Rules/CharacterGroupGenerator.cs

```diff
@@ -914,13 +1021,118 @@ public static void DistributeBooksAmongNarratorGroups(List<AuthorStats> authorSt
+								narratorPrefix, dramatizationPreferences);
+							break;
+
+						case CharacterVerseData.StandardCharacter.ExtraBiblical:
+							AddExtraBiblicalToGroupOrNarratorGroup(character.CharacterId, SectionHeadGroup, characterType,
+								narratorPrefix, dramatizationPreferences);
+							break;
+
+						case CharacterVerseData.StandardCharacter.Intro:
+							AddExtraBiblicalToGroupOrNarratorGroup(character.CharacterId, BookIntroductionGroup, characterType,
+								narratorPrefix, dramatizationPreferences);
+							break;
+					}
+				}
+			}
+
+			private void AddExtraBiblicalToGroupOrNarratorGroup(string characterId, CharacterGroup group,
+				CharacterVerseData.StandardCharacter characterType, string narratorPrefix, ProjectDramatizationPreferences dramatizationPreferences)
 			{
-				if (ExtraBiblicalGroup == null)
+				if (group != null)
+				{
+					group.CharacterIds.Add(characterId);
 					return;
-				ExtraBiblicalGroup.CharacterIds.AddRange(includedCharacterDetails.
-					Select(cd => cd.CharacterId).Where(c => CharacterVerseData.IsCharacterStandard(c, false) &&
-					!Groups.Any(g => g.CharacterIds.Contains(c))));
+				}
+
+				// add to a narrator group
+				var narratorGroup = GetExtraBiblicalToNarratorGroup(characterId, characterType, narratorPrefix, dramatizationPreferences);
+
+				if (narratorGroup != null)
+					narratorGroup.CharacterIds.Add(characterId);
+			}
+
+			private CharacterGroup GetExtraBiblicalToNarratorGroup(string characterId, CharacterVerseData.StandardCharacter characterType,
+				string narratorPrefix, ProjectDramatizationPreferences dramatizationPreferences)
+			{
+				var dramatizationOption = ExtraBiblicalMaterialSpeakerOption.Omitted;
+
+				switch (characterType)
+				{
+					case CharacterVerseData.StandardCharacter.BookOrChapter:
+						dramatizationOption = dramatizationPreferences.BookTitleAndChapterDramatization;
+						break;
+
+					case CharacterVerseData.StandardCharacter.ExtraBiblical:
+						dramatizationOption = dramatizationPreferences.SectionHeadDramatization;
+						break;
+
+					case CharacterVerseData.StandardCharacter.Intro:
+						dramatizationOption = dramatizationPreferences.BookIntroductionsDramatization;
+						break;
+				}
+
+				if (dramatizationOption == ExtraBiblicalMaterialSpeakerOption.Omitted)
+					return null;
+
+				var narratorId = narratorPrefix + characterId.Split(new[] { '-' }, 2)[1];
+				List<CharacterGroup> possibleGroups;
+
+				if (dramatizationOption == ExtraBiblicalMaterialSpeakerOption.Narrator)
+					possibleGroups = NarratorGroups.Where(g => g.CharacterIds.Contains(narratorId)).ToList();
+				else
+					possibleGroups = NarratorGroups.Where(g => !g.CharacterIds.Contains(narratorId)).ToList();
+
+				if (possibleGroups.Count == 0)
+					possibleGroups = NarratorGroups.Except(possibleGroups).ToList();
+
+				CharacterGroup narratorGroup = null;
+
+				// this should never happen, so throw an exception if it does
+				if (possibleGroups.Count == 0)
```

Comment:
This could be moved inside body of above if (possibleGroup.Count == 0) to save unnecessary re-check.